include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,1420 @@
+#define FDCAN_RXF0S_F0PI_SHIFT 16
+/* Rx FIFO 0 get index */
+#define FDCAN_RXF0S_F0GI_MASK 0x3F
+#define FDCAN_RXF0S_F0GI_SHIFT 8
+/* Rx FIFO 0 fill level */
+#define FDCAN_RXF0S_F0FL_MASK 0x7F
+#define FDCAN_RXF0S_F0FL_SHIFT 0
+
+/* --- FDCAN_RXF0A: FDCAN Rx FIFO 0 acknowledge register -------------------- */
+/* Rx FIFO 0 acknowledge index */
+#define FDCAN_RXF0A_F0AI_MASK 0x3F
+#define FDCAN_RXF0A_F0AI_SHIFT 0
+
+/* --- FDCAN_RXBC: FDCAN Rx buffer configuration register ------------------- */
+/* Rx buffer start address */
+#define FDCAN_RXBC_RBSA_MASK 0x3FFF
+#define FDCAN_RXBC_RBSA_SHIFT 2
+
+/* --- FDCAN_RXF1C: FDCAN Rx FIFO 1 configuration register ------------------ */
+/* FIFO 1Operation mode */
+#define FDCAN_RXF1C_F1OM BIT31
+/* Rx FIFO 1 Watermark */
+#define FDCAN_RXF1C_F1WM_MASK 0x7F
+#define FDCAN_RXF1C_F1WM_SHIFT 24
+/* Rx FIFO 1 size */
+#define FDCAN_RXF1C_F1S_MASK 0x7F
+#define FDCAN_RXF1C_F1S_SHIFT 16
+/* Rx FIFO 1 start address */
+#define FDCAN_RXF1C_F1SA_MASK 0x3FFF
+#define FDCAN_RXF1C_F1SA_SHIFT 2
+
+/* --- FDCAN_RXF1S: FDCAN Rx FIFO 1 status register ------------------------- */
+/* Debug message status */
+#define FDCAN_RXF1S_DMS_MASK 0x3
+#define FDCAN_RXF1S_DMS_SHIFT 30
+/* Rx FIFO 1 message Lost */
+#define FDCAN_RXF1S_RF1L BIT25
+/* Rx FIFO 1 Full */
+#define FDCAN_RXF1S_F1F BIT24
+/* Rx FIFO 1 put index */
+#define FDCAN_RXF1S_F1PI_MASK 0x3F
+#define FDCAN_RXF1S_F1PI_SHIFT 16
+/* Rx FIFO 1 get index */
+#define FDCAN_RXF1S_F1GI_MASK 0x3F
+#define FDCAN_RXF1S_F1GI_SHIFT 8
+/* Rx FIFO 1 fill level */
+#define FDCAN_RXF1S_F1FL_MASK 0x7F
+#define FDCAN_RXF1S_F1FL_SHIFT 0
+
+/* --- FDCAN_RXF1A: FDCAN Rx FIFO 1 acknowledge register -------------------- */
+/* Rx FIFO 1 acknowledge index */
+#define FDCAN_RXF1A_F1AI_MASK 0x3F
+#define FDCAN_RXF1A_F1AI_SHIFT 0
+
+/* --- FDCAN_RXESC: FDCAN Rx buffer element size configuration register ----- */
+/* Rx buffer data Field size: */
+enum fdcan_rxesc_rbds {
+	FDCAN_RXESC_RBDS_8		= 0x0,
+	FDCAN_RXESC_RBDS_12		= 0x1,
+	FDCAN_RXESC_RBDS_16		= 0x2,
+	FDCAN_RXESC_RBDS_20		= 0x3,
+	FDCAN_RXESC_RBDS_24		= 0x4,
+	FDCAN_RXESC_RBDS_32		= 0x5,
+	FDCAN_RXESC_RBDS_48		= 0x6,
+	FDCAN_RXESC_RBDS_64		= 0x7,
+	FDCAN_RXESC_RBDS_MASK	= 0x7,
+};
+#define FDCAN_RXESC_RBDS_SHIFT 8
+/* Rx FIFO 0 data Field size: */
+enum fdcan_rxesc_f1ds {
+	FDCAN_RXESC_F1DS_8		= 0x0,
+	FDCAN_RXESC_F1DS_12		= 0x1,
+	FDCAN_RXESC_F1DS_16		= 0x2,
+	FDCAN_RXESC_F1DS_20		= 0x3,
+	FDCAN_RXESC_F1DS_24		= 0x4,
+	FDCAN_RXESC_F1DS_32		= 0x5,
+	FDCAN_RXESC_F1DS_48		= 0x6,
+	FDCAN_RXESC_F1DS_64		= 0x7,
+	FDCAN_RXESC_F1DS_MASK	= 0x7,
+};
+#define FDCAN_RXESC_F1DS_SHIFT 4
+/* Rx FIFO 1 data Field size: */
+enum fdcan_rxesc_f0ds {
+	FDCAN_RXESC_F0DS_8		= 0x0,
+	FDCAN_RXESC_F0DS_12		= 0x1,
+	FDCAN_RXESC_F0DS_16		= 0x2,
+	FDCAN_RXESC_F0DS_20		= 0x3,
+	FDCAN_RXESC_F0DS_24		= 0x4,
+	FDCAN_RXESC_F0DS_32		= 0x5,
+	FDCAN_RXESC_F0DS_48		= 0x6,
+	FDCAN_RXESC_F0DS_64		= 0x7,
+	FDCAN_RXESC_F0DS_MASK	= 0x7,
+};
```

Comment:
same comment as earlier, these are all the same, it doesn't seem useful/necessary to repeat them

Reply:
done